sdk/servicebus/service-bus/test/internal/batchReceiver.spec.ts

```diff
@@ -1298,5 +1208,240 @@ describe("Batching Receiver", () => {
       }
       refreshConnectionCalled.should.be.greaterThan(0, "refreshConnection was not called.");
     });
+
+    it("returns messages if drain is in progress (receiveAndDelete)", async function(): Promise<
+      void
+    > {
+      // Create the sender and receiver.
+      await beforeEachTest("receiveAndDelete");
+
+      // Send a message so we can be sure when the receiver is open and active.
+      await sender.sendMessages(TestMessage.getSessionSample());
+
+      const messages1 = await receiver.receiveMessages(1, { maxWaitTimeInMs: 5000 });
+      should.equal(
+        messages1.length,
+        1,
+        "Unexpected number of received messages(before disconnect)."
+      );
+
+      const receiverContext = (receiver as ServiceBusSessionReceiverImpl)["_context"];
+      const batchingReceiver = (receiver as ServiceBusSessionReceiverImpl)["_messageSession"];
+
+      // Send a message so we have something to receive.
+      await sender.sendMessages(TestMessage.getSessionSample());
+
+      // We want to simulate a disconnect once the batching receiver is draining.
+      // We can detect when the receiver enters a draining state when `addCredit` is
+      // called while `drain` is set to true.
+      let didRequestDrain = false;
+      const addCredit = batchingReceiver["link"]!.addCredit;
+      batchingReceiver["link"]!.addCredit = function(credits) {
```

Comment:
Like all of our normal detached tests we have to rely on this `.idle()` hack.

Not for this PR (but maybe for the future) we should consider making these easier to unit test. It's pretty rare that we actually care about rhea since it's minimal in these cases and it's mostly about shutting down our own state.

Reply:
A bit complicated.. but definitely.. for a future PR.